server/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -240,9 +240,14 @@ private static void ensureUnionOutputsHaveCompatibleTypes(AnalyzedRelation left,
         for (int i = 0; i < leftOutputs.size(); i++) {
             Symbol leftOutput = leftOutputs.get(i);
             Symbol rightOutput = rightOutputs.get(i);
+
             DataType<?> leftType = leftOutput.valueType();
             DataType<?> rightType = rightOutput.valueType();
-            if (!DataTypes.isCompatibleType(leftType, rightType)) {
+
+            boolean isConvertable = rightType.precedes(leftType)
```

Comment:
The semantics are a bit specific to union and given that there are already some functions like `isCompatibleType` within `DataTypes` I think adding one more might make it more confusing rather than helping. I'd keep it here for now.

Reply:
Agree